src/workerd/api/global-scope.h

```diff
@@ -798,6 +773,17 @@ class ServiceWorkerGlobalScope: public WorkerGlobalScope {
     JSG_NESTED_TYPE(IdentityTransformStream);
     JSG_NESTED_TYPE(HTMLRewriter);
 
+    // Performance API
+    if (flags.getEnableGlobalPerformanceClasses() || flags.getEnableNodeJsPerfHooksModule()) {
```

Comment:
This needs a bit more work...

Reason: If I have enable_nodejs_perf_hooks on but also set `disable_global_performance_classes`, then `perf_hooks` should be available but these `globalThis` exports should not be.

I *think* what we actually want here is this:

This should only check the EnableGlobalPerformanceClasses flag:

```suggestion
    if (flags.getEnableGlobalPerformanceClasses()) {
```

Then, in the compatibility-date.capnp, `enable_global_performance_classes` should have an implied-by `enable_nodejs_perf_hooks_module` as of a specific date.

This would mean that when you enable perf_hooks, after the implied by date, you will get the `globalThis.Performance` *unless* you specifically have global performance turned off.

Reply:
Agreed. Applying these changes.